### DISCUSSION TITLE
Compatibility with SQS JSON Protocol

### DISCUSSION THREAD

**Discussion by nicholasgriffintn** [bbc/sqs-consumer#419] [Announcements]
AWS have made the announcement that they are switching from the XML protocol to JSON:

https://aws.amazon.com/about-aws/whats-new/2023/07/json-protocol-support-amazon-sqs/

This change has been made in version 3.327.0:

https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.327.0

In their announcement, AWS says that the protocol change "reduces end-to-end message processing latency by up to 23% and reduces application client side CPU and memory usage", so overall, this should be a good change.

We'll be downgrading our peer dependency requirement to support testing with this version in `7.2.2-sqs-json.1`, when that release is available, you'll be able to test the new protocol using that version

---

Update: AWS have now released the JSON protocol version in [v3.447.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.447.0), if you are having issues related to this, please downgrade to a previous version of the AWS SDK. There doesn't appear to be anything required from SQS Consumer in order to support this implementation.

**Comment by nicholasgriffintn** [comment]
Update: This version has now been released as the following canary version to preview (it should be noted that this version has security issues due to the AWS SDK preview for the JSON protocol requiring an older version of the SDK):

https://www.npmjs.com/package/sqs-consumer/v/7.2.2-sqs-json.1

Also, we have already found that as the response is different, communication with other AWS systems needs to be updated specifically to this version, as can be seen in the tests in this PR that are failing due to the conflict between the version that SQS Consumer is using for this version and the version that SQS Producer is using: https://github.com/bbc/sqs-consumer/pull/421

It should be noted that when AWS officially launch the new protocol outside of preview, this should be released as a major version, noting that all systems that use the AWS SDK need to be upgraded and checked for compatibility with the JSON protocol.

**Comment by mdrobny** [comment]
Looks like the JSON protocol is official and stable now https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-sqs-support-json-protocol/

**Comment by nicholasgriffintn** [comment]
Thanks! I did miss this. I'll have a look through next week, but presumably, it's just a case of updating the package.
